src/userSigner.ts

```diff
@@ -37,12 +37,15 @@ export class UserSigner implements ISigner {
     }
 
     private trySign(signable: ISignable) {
-        let signedBy = this.getAddress();
-        let bufferToSign = signable.serializeForSigning(signedBy);
+        let bufferToSign = signable.serializeForSigning();
         let signatureBuffer = this.secretKey.sign(bufferToSign);
         let signature = new Signature(signatureBuffer);
 
-        signable.applySignature(signature, signedBy);
+        this.doApplySignature(signable, signature);
+    }
+
+    protected doApplySignature(signable: ISignable, signature: ISignature) {
```

Comment:
can we keep the same `applySignature` name? I don't see an issue to be the same as on `ISignable`